src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrUpdater.cs

```diff
@@ -677,6 +684,87 @@ private string GetCurrentVersion(SourceMapping mapping)
         return version.Sha;
     }
 
+    private async Task CleanUpRemovedRepos(string? readmeTemplatePath, string? tpnTemplatePath)
+    {
+        var deletedRepos = _sourceManifest
+            .Repositories
+            .Where(r => _dependencyTracker.Mappings.FirstOrDefault(m => m.Name == r.Path) == null)
+            .ToList();
+
+        if (!deletedRepos.Any())
+        {
+            return;
+        }
+
+        using var vmrRepository = new Repository(_vmrInfo.VmrPath);
+
+        foreach (var repo in deletedRepos)
+        {
+            _logger.LogWarning("The mapping for {name} was deleted. Removing the repository from the VMR.", repo.Path);
+            var modifiedFiles = DeleteRepository(repo.Path);
+            
+            if(modifiedFiles.Any())
+            {
+                Commands.Stage(vmrRepository, modifiedFiles);
```

Comment:
I'd just stage `.` which stages any changed files and that way you wouldn't need to assemble the list of changes files but rather just return whether there are any changed.

Reply:
ok staging everything and committing once is the simplest option